Code/GraphMol/MolPickler.cpp

```diff
@@ -51,10 +51,16 @@ void streamRead(std::istream &ss, MolPickler::Tags &tag, int version) {
   if (version < 7000) {
     int32_t tmp;
     streamRead(ss, tmp, version);
+    if (tmp < 0 || tmp >= MolPickler::Tags::INVALID_TAG) {
+      throw MolPicklerException("Invalid tag found.");
```

Comment:
This will still have invalid tags as we have fewer than Tags::INVALID_TAG

I believe if you don't explicitly define INVALID_TAG=255 it will be one more than the maximum used tag value.

Reply:
Right, those are invalid but it's not undefined behavior if the value doesn't exceed the maximum specified for the enum:
`If the underlying type is not fixed and the source value is out of range, the result is unspecified (until C++17)undefined (since C++17).`